<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="MÃ¥ns Magnusson">
        
        <link rel="shortcut icon" href="../img/favicon.ico">

	<title>Genetic Models - genmod docs</title>

        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="..">genmod docs</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="..">Home</a>
                    </li>
                
                
                
                    <li class="active">
                        <a href="./">Genetic Models</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Commands <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../commands/annotate-variants/">Annotating Variants</a>
</li>

                        
                            
<li >
    <a href="../commands/annotate-models/">Annotate Genetic Models</a>
</li>

                        
                            
<li >
    <a href="../commands/score-variants/">Score Variants</a>
</li>

                        
                            
<li >
    <a href="../commands/score-compounds/">Score Compounds</a>
</li>

                        
                            
<li >
    <a href="../commands/filter-variants/">Filter Variants</a>
</li>

                        
                            
<li >
    <a href="../commands/sort-variants/">Sort Variants</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="..">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../commands/annotate-variants/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
                    <li>
                        <a href="https://github.com/moonso/genmod">
                            
                                <i class="fa fa-github"></i>
                            
                            GitHub
                        </a>
                    </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#conditions-for-genetic-models">Conditions for Genetic Models</a></li>
        
            <li><a href="#short-explanation-of-genotype-calls-in-vcf-format">Short explanation of genotype calls in VCF format:</a></li>
        
            <li><a href="#autosomal-recessive">Autosomal Recessive</a></li>
        
            <li><a href="#autosomal-dominant">Autosomal Dominant</a></li>
        
            <li><a href="#autosomal-compound-heterozygote">Autosomal Compound Heterozygote</a></li>
        
            <li><a href="#x-linked-dominant">X-Linked Dominant</a></li>
        
            <li><a href="#x-linked-recessive">X Linked Recessive</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h2 id="conditions-for-genetic-models">Conditions for Genetic Models</h2>
<h3 id="short-explanation-of-genotype-calls-in-vcf-format">Short explanation of genotype calls in VCF format:</h3>
<p>Since we only look at humans, that are diploid, the genotypes represent what we see on both alleles in a single position.
<code>0</code> represents the reference sequence, <code>1</code> is the first of the alternative alleles, <code>2</code> second alternative and so on.
If no phasing has been done the genotype is an unordered pair on the form <code>x/x</code>, so <code>0/1</code> means that the individual is heterozygote in this given position with the reference base on one of the alleles and the first of the alternatives on the other.
<code>2/2</code> means that we see the second of the alternatives on both alleles.
Some chromosomes are only present in one copy in humans, here it is allowed to only use a single digit to show the genotype. A <code>0</code> would mean reference and <code>1</code> first of alternatives.</p>
<p>If phasing has been done the pairs are not unordered anymore and the delimiter is then changed to '|', so one can be heterozygote in two ways; <code>0|1</code> or <code>1|0</code>.</p>
<h3 id="autosomal-recessive">Autosomal Recessive</h3>
<p>For this model individuals can be carriers so healthy individuals can be heterozygous. Both alleles need to have the variant for an individual to be sick so a healthy individual can not be homozygous alternative and a sick individual <em>has</em> to be homozygous alternative.</p>
<ul>
<li>Affected individuals have to be homozygous alternative (hom. alt.)</li>
<li>Healthy individuals cannot be hom. alt.</li>
<li>Variant is considered <em>de novo</em> if both parents are genotyped and do not carry the variant</li>
<li>The variants following this modell will be annotated with <code>AR_hom</code>, or <code>AR_hom_dn</code> for de novo</li>
</ul>
<h3 id="autosomal-dominant">Autosomal Dominant</h3>
<ul>
<li>Affected individuals have to be heterozygous (het.)</li>
<li>Healthy individuals cannot have the alternative variant</li>
<li>Variant is considered <em>de novo</em> if both parents are genotyped and do not carry the variant</li>
<li>The variants following this modell will be annotated with <code>AD</code>or <code>AD_dn</code></li>
</ul>
<h3 id="autosomal-compound-heterozygote">Autosomal Compound Heterozygote</h3>
<p>This model includes pairs of exonic variants that are present within the same gene.
<strong>The default behaviour of GENMOD is to look for compounds only in exonic/canonical splice sites</strong>.
The reason for this is that since some genes have huge intronic regions the data will be drowned in compound pairs.
If the user wants all variants in genes checked use the flag -gene/--whole_gene.</p>
<ol>
<li>
<p>Non-phased data:</p>
<ul>
<li>Affected individuals have to be het. for both variants</li>
<li>Healthy individuals can be het. for one of the variants but cannot have both variants</li>
<li>Variant is considered <em>de novo</em> if only one or no variant is found in the parents</li>
</ul>
</li>
<li>
<p>Phased data:</p>
<ul>
<li>All affected individuals have to be het. for both variants <strong>and</strong> the variants has to be on two different alleles</li>
<li>Healthy individuals can be heterozygous for one but cannot have both variants</li>
<li>If only one or no variant is found in parents it is considered <em>de novo</em></li>
</ul>
</li>
</ol>
<p>Variant following this patters will be annotated with <code>AR_comp</code> or <code>AR_comp_dn</code></p>
<h3 id="x-linked-dominant">X-Linked Dominant</h3>
<p>These traits are inherited on the x-chromosome, of which men have one allele and women have two.</p>
<ul>
<li>Variant has to be on chromosome X</li>
<li>Affected individuals have to be het. or hom. alt.</li>
<li>Healthy males cannot carry the variant</li>
<li>Healthy females can carry the variant (because of X inactivation)</li>
<li>If sex is male the variant is considered <em>de novo</em> if mother is genotyped and does not carry the variant</li>
<li>If sex is female variant is considered <em>de novo</em> if none of the parents carry the variant</li>
<li>The variants following this modell will be annotated with <code>XD</code>or <code>XD_dn</code></li>
</ul>
<h3 id="x-linked-recessive">X Linked Recessive</h3>
<ul>
<li>Variant has to be on chromosome X</li>
<li>Affected males have to be het. or hom. alt. (het is theoretically not possible in males, but can occur due to Pseudo Autosomal Regions).</li>
<li>Affected females have to be hom. alt.</li>
<li>Healthy females cannot be hom. alt.</li>
<li>Healthy males cannot carry the variant</li>
<li>If sex is male the variant is considered <em>de novo</em> if mother is genotyped and does not carry the variant</li>
<li>If sex is female variant is considered <em>de novo</em> if not both parents carry the variant</li>
<li>The variants following this modell will be annotated with <code>XR</code>or <code>XR_dn</code></li>
</ul></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../js/jquery-1.10.2.min.js"></script>
        <script src="../js/bootstrap-3.0.3.min.js"></script>
        <script src="../js/highlight.pack.js"></script>
        <script>var base_url = '..';</script>
        <script data-main="../mkdocs/js/search.js" src="../mkdocs/js/require.js"></script>
        <script src="../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
